kolibri/plugins/coach/assets/src/views/class-selector/index.vue

```diff
@@ -0,0 +1,99 @@
+<template>
+
+  <div>
+    <router-link :to="classListPage"><h2>{{ $tr('allClasses') }}</h2></router-link>
+    <span class="seperator">&#62;</span>
+    <ui-select
+      :name="$tr('selectClass')"
+      :label="$tr('selectClass')"
+      :value="currentClass"
+      :options="classOptions"
+      @change="changeClass"
+      class="class-selector"
+    />
+  </div>
+
+</template>
+
+
+<script>
+
+  const orderBy = require('lodash/orderBy');
+  const PageNames = require('../../constants').PageNames;
+
+
+  module.exports = {
+    $trNameSpace: 'classSelector',
+    $trs: {
+      allClasses: 'All classes',
+      selectClass: 'Class',
+    },
+    components: {
+      'ui-select': require('keen-ui/src/UiSelect'),
+      'ui-icon': require('keen-ui/src/UiIcon'),
+    },
+    props: {
+      classes: {
+        type: Array,
+        required: true,
+      },
+      currentClassId: {
+        type: String,
+        required: true,
+      },
+    },
+    computed: {
+      sortedClasses() {
+        return orderBy(
+          this.classes,
+          [classroom => classroom.name.toUpperCase()],
+          ['asc']
+        );
+      },
+      classOptions() {
+        return this.sortedClasses.map(classroom => ({ label: classroom.name, id: classroom.id }));
+      },
+      currentClass() {
+        return this.classOptions.find(classroom => classroom.id === this.currentClassId);
+      },
+      classListPage() {
+        return PageNames.CLASS_LIST;
+      },
+    },
+    methods: {
+      changeClass(classSelected) {
+        this.$router.push({
+          params: { classId: classSelected.id },
```

Comment:
I'm not used to seeing a push without a pageName set. Does this work as expected for all sub-pages of all sections?

Consider for example a report view of an exam, which would only exist in one class.